app/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -100,6 +101,10 @@
     private Snackbar snackbar;
     private boolean duplicateCheckPassed = false;
 
+    private String nearbyTitle;
+    private String nearbyDesc;
+    private boolean isNearbyUpload = false;
```

Comment:
This activity doesn't need to know about the title, description and whether this is a nearby upload.  These should live in the `SingleUploadFragment` to reduce coupling + all the nasty casting.